### PR TITLE
chore(linter): require a test label for CLI changes

### PR DESCRIPTION
**Pull request by rix0rrr**
Require a label a human must add for PRs that change CLI code. The linter will give instructions on how to push the code to the right pipeline branch.

Also add strong typing to the linter, there were a couple of bugs in there because of the `any`s.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: tools/@aws-cdk/prlint/lint.ts [hunk 1]
```diff
@@ -330,25 +363,35 @@ function hasLabel(pr: any, labelName: string): boolean {
...
+function noCliChanges(pr: GitHubPr, files: GitHubFile[]): TestResult {
+  const branch = `pull/${pr.number}/head`;
+
+  const cliCodeChanged = files.some(f => f.filename.toLowerCase().includes('/packages/aws-cdk/lib/') && f.filename.endsWith('.ts'));
+
+  return TestResult.fromFailure(
+    cliCodeChanged,
+    `CLI code has changed. A maintainer must run the code through the testing pipeline (git fetch origin ${branch} && git push -f origin FETCH_HEAD:test-main-pipeline), then add the '${Exemption.CLI_INTEG_TESTED}' label when the pipeline succeeds.`);
+}
+
```
Comment (Naumel) [inline tools/@aws-cdk/prlint/lint.ts:394]: A test would be swell

**Review by Naumel — APPROVED** [review]
lgtm

**Comment by Naumel** [comment]
Mentioning #22175 here due to the output

File: tools/@aws-cdk/prlint/lint.ts (hunk 1 above)
Comment (rix0rrr) [reply tools/@aws-cdk/prlint/lint.ts]: Yep, added it.

**Review by Naumel — APPROVED** [review]

**Pull request merged**